phc/services/tools.py

```diff
@@ -0,0 +1,281 @@
+"""A Python Module for Files"""
+
+import os
+import math
+import backoff
+from phc.base_client import BaseClient
+from phc import ApiResponse
+from urllib.parse import urlencode
+from urllib.request import urlretrieve
+from phc.errors import ApiError
+
+
+class Tools(BaseClient):
+    """Provides acccess to PHC tools registry
+
+    Parameters
+    ----------
+    session: phc.Session
+        The PHC session
+    run_async: bool
+        True to return promises, False to return results (default is False)
+    timeout: int
+        Operation timeout (default is 30)
+    trust_env: bool
+        Get proxies information from HTTP_PROXY / HTTPS_PROXY environment variables if the parameter is True (False by default)
+    """
+
+    def create(
+        self,
+        name: str,
+        description: str,
+        access: str,
+        version: str,
+        tool_class: str,
+        source: str,
+        labels: str = None,
+    ) -> ApiResponse:
+        """Create a tool.
+
+        Parameters
+        ----------
+        name: str
+            The name to give to the tool
+        description: str
+            A description of the tool
+        access: str
+            The access level given to the tool [PRIVATE, ACCOUNT, PHC, PUBLIC]
+        version: str
+            The initial version of the tool
```

Comment:
Does this version parameter have requirements? I'd love to have a bit more validation on these parameters in general if possible (e.g. `access` or `tool_class`). Checkout what we've done with runtime and compile-time validation on [genomic_expression.py](https://github.com/lifeomic/phc-sdk-py/blob/67b706e2f2a206cfd5de777f4a93904538a62a92/phc/easy/omics/options/genomic_expression.py) (using [pydantic](https://pydantic-docs.helpmanual.io/)).

(If you don't have bandwidth to do this at the moment, just let me know, and I'll be happy to jump in to help.)

Reply:
I'll add in that pattern, I was depending upon the api calls doing that for us, but if we have a pattern of doing that here, that's fine.

Reply:
And version can be any string.